ilastik_scripts/ilastik_startup.py

```diff
@@ -138,9 +143,59 @@ def find_library(name):
     ctypes.util.find_library = find_library
 
 
+def fix_ssl() -> None:
+    """
+    Paths to cert.pem and the CA certificates dir are hard-coded into openssl during the build process
+    of our installer packages.
+    It has some special logic for Windows, and there the paths don't seem to be an issue.
+    On Linux and macOS, the build machine's path is written (/opt/conda/envs/ilastik-release/ssl/),
+    which does not exist on the end user's machine.
+    Being unable to find the cert files means aiohttp cannot connect via https. Currently only
+    zarr.storage.FSStore depends on this.
+    We override the broken paths using the env vars that openssl listens to. Better would be if we can
+    find a way to write the correct paths during build.
+    """
+    if platform.system() == "Windows":
+        return
+
+    ssl_paths = ssl.get_default_verify_paths()
+    if os.path.isfile(ssl_paths.openssl_cafile):
+        # All good then I guess
+        return
+
+    def check_parent_for_dir_recursive(path: pathlib.Path, dir_name: str) -> Optional[pathlib.Path]:
+        if path.is_dir() and any(p.name == dir_name for p in path.iterdir()):
+            return path
+        if path.parent == path:
+            raise RecursionError(f"Could not find {dir_name} directory")
```

Comment:
This short-cuts to the actual RecursionError that I was getting before adding this if-clause to exit the recursion. In that state, the function just keeps recursing because when `path` is the FS root, `path.parent` is also the FS root.

I had it returning None in the previous commit and found the code unreadable that way; the try-except phrasing seemed more expressive.

Reply:
Wouldn't something like this be easier to work with instead of recursion and nested exception handling?

```python
# ... explain why PWD/OLDPWD is here ...
candidate_ssl_parents = [
    *pathlib.Path(__file__).parents,
    os.environ.get("PWD"),
    os.environ.get("OLDPWD"),
]
candidate_ssl_parents = list(map(pathlib.Path, filter(None, parents)))

for parent in parents:
    if any(
        (p / "ssl/cert.pem").is_file() and (p / "ssl/certs").is_dir()
        for p in parent.iterdir()
    ):
        logger.info(...)
        ...
        break
else:
    logger.warning(...)
```